### PR TITLE
Test/add cmd golden file tests

### PR DESCRIPTION
**Pull request by tim775**

### CODE REVIEW

File: cmd/infracost/cmd_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/infracost/infracost/cmd/infracost"
+	"github.com/infracost/infracost/internal/config"
+	"github.com/infracost/infracost/internal/testutil"
+	"github.com/stretchr/testify/require"
+	"path/filepath"
+	"regexp"
+	"testing"
+)
+
+var timestampRegex = regexp.MustCompile(`(\d{4})-(\d{2})-(\d{2})(T| )(\d{2}):(\d{2}):(\d{2}(?:\.\d*)?)((-(\d{2}):(\d{2})|Z| [A-Z]{3})?)`)
```
Comment (aliscott) [inline cmd/infracost/cmd_test.go:16]:
```suggestion
var timestampRegex = regexp.MustCompile(`(\d{4})-(\d{2})-(\d{2})(T| )(\d{2}):(\d{2}):(\d{2}(?:\.\d*)?)(([\+-](\d{2}):(\d{2})|Z| [A-Z]{3})?)`)
```

It was working with `2021-08-30T15:04:15-01:00` (with -) but not `2021-08-30T15:04:15+01:00` (with +)

**Review by aliscott — APPROVED** [review]
This looks great.

I see the refactors for printing to the cmd's out instead of stdout. Do we want to do a similar thing for stderr so we can have golden tests for the error outputs as well?

File: internal/config/config.go [hunk 2]
```diff
@@ -45,6 +46,11 @@ type Config struct {
 	ShowSkipped   bool       `yaml:"show_skipped,omitempty" ignored:"true"`
 	SyncUsageFile bool       `yaml:"sync_usage_file,omitempty" ignored:"true"`
 	Fields        []string   `yaml:"fields,omitempty" ignored:"true"`
+
+	// for testing
+	EventsDisabled       bool
```
[the remark this answers is not in the document]
Comment (tim775) [reply internal/config/config.go:51]: Yeah that would be ok.  Is there a good way to detect that we're in a test?

[the remark this answers is not in the document]
Comment (aliscott) [reply internal/config/config.go:51]: We have https://github.com/infracost/infracost/blob/master/internal/config/config.go#L166

**Comment by tim775** [comment]
> I see the refactors for printing to the cmd's out instead of stdout. Do we want to do a similar thing for stderr so we can have golden tests for the error outputs as well?

I had looked into this and decided it was too much of a pain to pass the cmd context down through all the levels, but after a second look I added code so we're capturing output from the ui.Print* methods.  I think this is probably a good compromise.  Thanks for the prod!

**Comment by aliscott** [comment]
Yeah that works! And in the future we should probably refactor everywhere that uses `fmt.Fprintln(os.Stderr, ...` to use the `ui.*` functions.

**Comment by tim775** [comment]
@kevinkengne just a heads up that I added these tests.  Thanks for the suggestion!

**Pull request merged**